lienzo-core/src/main/java/com/ait/lienzo/client/widget/panel/impl/ScrollablePanel.java

```diff
@@ -299,6 +330,107 @@ private void onScroll() {
             final double sv = getVerticalScrollRate();
             applyScrollRateToLayer(sh, sv);
         }
+        isScrolling = false;
+    }
+
+    private void onPrimitiveDragStart(IPrimitive<?> primitive) {
+        final BoundingBox primitiveBoundingBox = new BoundingBox();
+        for (Point2D point2D : primitive.getComputedBoundingPoints()) {
+            primitiveBoundingBox.add(point2D.getX(), point2D.getY());
+        }
+
+        final BoundingBox visibleBoundingBox = new BoundingBox();
+        final Bounds visibleBounds = getVisibleBounds();
+        visibleBoundingBox.add(visibleBounds.getX(), visibleBounds.getY());
+        visibleBoundingBox.add(visibleBounds.getX() + visibleBounds.getWidth(), visibleBounds.getY() + visibleBounds.getHeight());
+
+        double padding = getBoundsProvider().getPadding();
+
+        double primMinX = primitiveBoundingBox.getMinX();
+        double primMaxX = primitiveBoundingBox.getMaxX();
+
+        double visMinX = visibleBoundingBox.getMinX();
+        double visMaxX = visibleBoundingBox.getMaxX();
+
+        if (primMinX - padding < visMinX) {
+            dragDifferenceX = visMinX - primMinX;
+        } else if (primMaxX + padding > visMaxX) {
+            dragDifferenceX = visMaxX - primMaxX;
+        }
+    }
+
+    private void onPrimitiveDragUpdate(IPrimitive<?> primitive) {
+        final BoundingBox primitiveBoundingBox = new BoundingBox();
+        for (Point2D point2D : primitive.getComputedBoundingPoints()) {
```

Comment:
Calculating BoundingBox'es or BondingPoints is expensive, and this call here is done on each drag update (for every pixel), so thsi method will be called several times when just dragging some primitive shape a little bit.Could be possible to avoid it in some way? 
If you just need the new primitive location (while dragging), the idea is to keep the inital x / y on dragStart, and then every dragUpdate call just need to increase/decrease both x/y properly, this is the way lienzo core works and avoids even accessing the primitive shape in memoty to obtain location, this is safe and fast, you can looks for examples in classes in lienzo.